service/s3control/internal/customizations/process_arn_resource.go

```diff
@@ -110,6 +113,15 @@ func (m *processARNResource) HandleSerialize(
 			return out, metadata, fmt.Errorf("error updating arnable field while serializing")
 		}
 
+		if !s3shared.IsClonedInput(ctx) {
+			in.Parameters, err = m.CopyInput(in.Parameters)
+			if err != nil {
+				return out, metadata, fmt.Errorf("error creating a copy of input while processing arn")
+			}
+			// set copy input key on context
+			ctx = s3shared.SetClonedInputKey(ctx, true)
+		}
```

Comment:
nit: You could move the clone action to immediately before the switch statement. At worst if the user provided an invalid ARN resource we would have made an unnecessary copy. But it results in not needing to duplicate this between resource types.